app/src/components/widgets/charts/launchExecutionAndIssueStatistics/issueStatisticsChart.jsx

```diff
@@ -0,0 +1,271 @@
+import { Legend } from './launchExecutionAndIssuesChartLegend';
+import { LaunchExecutionAndIssueStatisticsTooltip } from './launchExecutionAndIssueStatisticsTooltip';
+import { getPercentage, getDefectItems } from './chartUtils';
+
+const cx = classNames.bind(styles);
+
+const messages = defineMessages({
+  statistics$defects$product_bug: {
+    id: 'FilterNameById.statistics$defects$product_bug',
+    defaultMessage: 'Product bug',
+  },
+  statistics$defects$automation_bug: {
+    id: 'FilterNameById.statistics$defects$automation_bug',
+    defaultMessage: 'Automation bug',
+  },
+  statistics$defects$system_issue: {
+    id: 'FilterNameById.statistics$defects$system_issue',
+    defaultMessage: 'System issue',
+  },
+  statistics$defects$no_defect: {
+    id: 'FilterNameById.statistics$defects$no_defect',
+    defaultMessage: 'No defect',
+  },
+  statistics$defects$to_investigate: {
+    id: 'FilterNameById.statistics$defects$to_investigate',
+    defaultMessage: 'To investigate',
+  },
+  ofTestCases: {
+    id: 'Widgets.ofTestCases',
+    defaultMessage: 'of test cases',
+  },
+});
+
+@injectIntl
+export class IssueStatisticsChart extends Component {
+  static propTypes = {
+    intl: intlShape.isRequired,
+    widget: PropTypes.object.isRequired,
+    isPreview: PropTypes.bool,
+    container: PropTypes.instanceOf(Element).isRequired,
+    observer: PropTypes.object.isRequired,
+  };
+
+  static defaultProps = {
+    isPreview: false,
+    height: 0,
+  };
+
+  state = {
+    isConfigReady: false,
+  };
+
+  componentDidMount() {
+    this.props.observer.subscribe('widgetResized', this.resizeIssuesChart);
+    this.getConfig();
+  }
+  componentWillUnmount() {
+    this.issuesNode.removeEventListener('mousemove', this.getCoords);
+    this.props.observer.unsubscribe('widgetResized', this.resizeIssuesChart);
+  }
+
+  onIssuesChartCreated = (chart, element) => {
+    this.issuesChart = chart;
+    this.issuesNode = element;
+    if (!this.props.widget.content.result || this.props.isPreview) {
+      return;
+    }
+
+    d3
+      .select(chart.element)
+      .select('.c3-chart-arcs-title')
+      .attr('dy', -15)
+      .append('tspan')
+      .attr('dy', 30)
+      .attr('x', 0)
+      .attr('fill', '#666')
+      .text('ISSUES');
+
+    this.issuesNode.addEventListener('mousemove', this.getCoords);
+  };
+
+  onIssuesMouseOut = () => {
+    this.issuesChart.revert();
+  };
+
+  onIssuesMouseOver = (id) => {
+    this.issuesChart.focus(id);
+  };
+
+  onIssuesClick = (id) => {
+    this.issuesChart.toggle(id);
+  };
+
+  getConfig = () => {
+    const { widget, container } = this.props;
+
+    this.height = container.offsetHeight;
+    this.width = container.offsetWidth;
+    this.noAvailableData = false;
+    const defectTypesChartData = {};
+    const defectTypesChartDataOrdered = [];
+    const defectTypesChartColors = {};
+    const values = widget.content.result[0].values;
+    Object.keys(values).forEach((key) => {
```

Comment:
Looks like `values.reduce` will be better to fit here for fill `defectTypesChartData` object.

Reply:
could you elaborate one this a bit more? values is an object, it doesn't have a reduce function

Reply:
Oh, it's my mistake, you right. Let's leave it as is.